core/src/test/kotlin/ch/ergon/dope/buildTest/SubQueryTest.kt

```diff
@@ -21,9 +29,58 @@ class SubQueryTest {
 
         val actual: String = create
             .select(someStringField())
-            .from(create.selectAsterisk().from(someBucket()).alias("asdf"))
+            .from(create.selectAsterisk().from(someBucket()).queryAlias("asdf"))
             .build().queryString
 
         assertEquals(expected, actual)
     }
+
+    @Test
+    fun `should support sub select with in array`() {
+        val expected = "SELECT TRUE IN (SELECT RAW FALSE FROM `other`) FROM `someBucket`"
+
+        val actual: String = create
+            .select(
+                TRUE.inArray(
+                    create.selectRaw(FALSE).from(someBucket("other")),
+                ),
+            ).from(
+                someBucket(),
+            ).build().queryString
+
+        assertEquals(expected, actual)
+    }
+
+    @Test
+    fun `should support sub select in EXISTS`() {
+        val expected = "SELECT EXISTS (SELECT * FROM `someBucket`)"
+
+        val actual = create.select(
+            exists(someSelectClause().from(someBucket())),
+        ).build().queryString
+
+        assertEquals(expected, actual)
+    }
+
+    @Test
+    fun `should support sub select in ARRAY_LENGTH`() {
+        val expected = "SELECT ARRAY_LENGTH((SELECT * FROM `someBucket`))"
+
+        val actual = create.select(
+            arrayLength(someSelectClause().from(someBucket())),
+        ).build().queryString
+
+        assertEquals(expected, actual)
+    }
+
+    @Test
+    fun `should support sub select in Equals`() {
+        val expected = "SELECT `stringField` = (SELECT RAW `stringField` FROM `someBucket`)[0]"
+
+        val actual = create.select(
+            someStringField().isEqualTo(create.selectRaw(someStringField()).from(someBucket()).get(0)),
+        ).build().queryString
+
+        assertEquals(expected, actual)
+    }
```

Comment:
How do parameters in subqueries work? Are there tests for that kind of behaviour?

Reply:
They should work the same way as before.
I added a test